src/node/miner.cpp

```diff
@@ -520,14 +623,9 @@ std::unique_ptr<CBlockTemplate> WaitAndCreateNewBlock(ChainstateManager& chainma
          * We'll also create a new template if the tip changed during this iteration.
          */
         if (options.fee_threshold < MAX_MONEY || tip_changed) {
-            auto new_tmpl{BlockAssembler{
-                chainman.ActiveChainstate(),
-                mempool,
-                assemble_options}
-                              .CreateNewBlock()};
-
+            auto new_tmpl{block_template_cache->GetBlockTemplate(assemble_options, std::chrono::seconds{0}).first};
```

Comment:
In commit "node: add block template cache to miner" (4de7887ca47197654f480471db04750d44e405de)

Would be good to note that `0` interval means this will always generate a new template, not use a cached one.